### ISSUE TITLE
Poor activation UX for meta queries

### ISSUE THREAD

**Issue by danielbachhuber** [pantheon-systems/solr-power#188]
As a Solr Power user, I was confused by the fact that I needed to whitelist my post type's meta keys in order to have them sent to the index. Because I was confused, Solr Power didn't work for one of my use cases until I discovered the option, saved the appropriate values, and triggered a reindex.

**Comment by ataylorme** [comment]
> I was confused by the fact that I needed to whitelist my post type's meta keys in order to have them sent to the index

@danielbachhuber do you think this can be solved with documentation?

**Comment by danielbachhuber** [comment]
Potentially, although I don't think most people read the docs fully. It's a confusing user experience to have custom taxonomy data handled automatically, but meta values ignored.

**Comment by ataylorme** [comment]
Agreed but not all meta values for a post type will need to be indexed. In the books example there might be Yoast SEO meta data, etc. Taxonomies tend not to be as cluttered.

Indexed meta also gets turned into facets and doing so automatically could create a search widget with a lot of irrelevant options.

Any thoughts on how to avoid indexing a bunch of meta that's not needed?

**Comment by danielbachhuber** [comment]
> Agreed but not all meta values for a post type will need to be indexed. In the books example there might be Yoast SEO meta data, etc. Taxonomies tend not to be as cluttered.
> 
> Indexed meta also gets turned into facets and doing so automatically could create a search widget with a lot of irrelevant options.

Makes sense.

> Any thoughts on how to avoid indexing a bunch of meta that's not needed?

There's a core API for explicitly marking meta as public vs. private, but it tends to not be frequently used.

I'd look at the Elastic Search plugins to see how they've solved the problem.

One approach could be to have an obtrusive nag until settings have been saved once, and inject the document into that user experience.
